### ISSUE TITLE
[google_maps] Randomly crashes with java.lang.NullPointerException

### ISSUE THREAD

**Issue by marianhlavac** [flutter/flutter#79490]
Flutter application suddenly crashed on the initial render of the google map, with unhandled exception:

**`java.lang.NullPointerException: Attempt to get length of null array at java.nio.ByteBufferAsShortBuffer.put(ByteBufferAsShortBuffer.java:121)`**

## Steps to Reproduce

Issue is not reproducible on minimal example, not even in the same Flutter application the bug actually happened. Restarting the app fixed the issue.

What happened before and after:

* App crashed on hot reload when developing (not sure if relevant)
* I decided to terminate the process and performed rebuild and re-run
* App immediately crashed when map loaded

## Details

Re-running the application with verbose option won't help, as the application loaded fine on second re-run.

There are no relevant analysis errors. (we just have snackbar deprecation warnings from upgrading to 2.0)

<details>
  <summary>Logs</summary>

```
E/AndroidRuntime(31038): FATAL EXCEPTION: GLThread 15083
E/AndroidRuntime(31038): Process: com.***.app.bundle.name, PID: 31038
E/AndroidRuntime(31038): java.lang.NullPointerException: Attempt to get length of null array
E/AndroidRuntime(31038): 	at java.nio.ByteBufferAsShortBuffer.put(ByteBufferAsShortBuffer.java:121)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.gl.buffer.c.m(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):2)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.gl.buffer.c.l(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):1)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.a.t(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):6)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.bg.E(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):4)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.bg.t(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):5)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.bs.c(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):144)
E/AndroidRuntime(31038): 	at com.google.maps.api.android.lib6.gmm6.vector.an.run(:com.google.android.gms.dynamite_mapsdynamite@210915066@21.09.15 (100406-0):49)
[MapBloc.MESSAGE] MapCreated
I/Process (31038): Sending signal. PID: 31038 SIG: 9
Lost connection to device.
```

I've copied complete logcat from the device roughly around the time the app started, was running and crashed. I've just redacted our bundle name, so it accidentally don't pop up in google search.

https://gist.github.com/marianhlavac/83018ac57a1de64c8d1ea3de8501be59

It's mostly unusable mess, but it might give some insight, if someone knows what he's looking for.

```
[✓] Flutter (Channel stable, 2.0.3, on Mac OS X 10.15.7 19H114 darwin-x64, locale en-GB)
    • Flutter version 2.0.3 at /usr/local/Caskroom/flutter/1.22.5/flutter
    • Framework revision 4d7946a68d (13 days ago), 2021-03-18 17:24:33 -0700
    • Engine revision 3459eb2436
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/marian/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.10.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.54.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.20.0

[✓] Connected device (2 available)
    • Redmi 7 (mobile) • df7268bc • android-arm64  • Android 9 (API 28)
    • Chrome (web)     • chrome   • web-javascript • Google Chrome 89.0.4389.114

• No issues found!
```

</details>

---

I understand that the given amount of information isn't sufficient to resolve the issue. But since it's not reproducible, it's all I have for now.

I'm willing to provide more information on how we use Google map in our app, or to perform some tests on the Android device, if it becomes a suspect. Should I collect any specific information, next time the issue occurs again?

Thanks.

**Comment by darshankawar** [comment]
@marianhlavac 
Thanks for the report. On your first point, `App crashed on hot reload when developing (not sure if relevant)`, what kind of crash did it throw ? Was it the same crash or different ? Is there a way to trigger this crash again and provide it ?

`App immediately crashed when map loaded`. Can you please elaborate on the scenario in which you are loading the map ? Does it load after triggering some other action or as soon as the app is launched ? If possible, can you provide a use case around it which may possibly trigger the crash ?

I also think that since the crash is coming from native `java.lang.NullPointerException: Attempt to get length of null array at java.nio.ByteBufferAsShortBuffer.put(ByteBufferAsShortBuffer.java:121)` and doesn't point to any file related to the plugin, it could be an issue from Android side.  See this similar issue for your reference, https://stackoverflow.com/questions/27253524/attempt-to-get-length-of-null-array-in-google-maps-v2-android-app

**Comment by marianhlavac** [comment]
@darshankawar Thanks for the quick reply.

> what kind of crash did it throw?

The crash is a segmentation fault that happens to me from time to time when hot reloading.

```F/libc    (18572): Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x340117 in tid 18656 (1.ui), pid 18572 (****)```

> I also think that since the crash is coming from native and doesn't point to any file related to the plugin

I agree, I stumbled upon the same link on stack overflow you've sent, but never noticed the other link mentioned in one of the answers leading to this issue:
https://code.google.com/p/gmaps-api-issues/issues/detail?id=5100#c34

It seems like the exact issue I've encountered and other people describe it the same - it happens only sometimes. 

**It really looks like native issue. I will continue the discussion in the [google issue tracker thread](https://code.google.com/p/gmaps-api-issues/issues/detail?id=5100#c34), and for the crashing hot reload I will collect more data, look for other existing issues or open a new issue dedicated to the hot reload problems.**

Thanks for the assistance.
